homeassistant/components/emoncms/sensor.py

```diff
@@ -196,30 +195,6 @@ def __init__(
             self._attr_device_class = SensorDeviceClass.PRESSURE
             self._attr_state_class = SensorStateClass.MEASUREMENT
 
-        if self._value_template is not None:
-            self._attr_native_value = (
-                self._value_template.render_with_possible_json_value(
-                    elem["value"], STATE_UNKNOWN
-                )
-            )
-        elif elem["value"] is not None:
-            self._attr_native_value = round(float(elem["value"]), DECIMALS)
-        else:
-            self._attr_native_value = None
```

Comment:
I thought to call `self.update()` at the end of the `__init__` of the `EmonCmsSensor` class
is it OK for you ?

For an upcoming PR, I think also  :
- there is not benefit to have feedtag and feedname in the identifier,
- there is one to have feed tag in the friendly name (`self._attr_name`) as users will identify more easily the feeds
- the code in the `__init__` which fixes `self._attr_name` could be in the update method, because if a user modify a tag or a feed name, it will be reported in home assistant, which is not the case with the actual implementation

Reply:
You shouldn't call `self.update()` inside `__init__`, as it has already been called once in `setup_platform`
This is why I think you should split `get data` and `set attributes` to avoid duplicate code

Reply:
in `setup_platform`, it is the `update` method of the `EmonCmsData` class which is called, that's why I thought it could be possible to call the `update` method of `EmonCmsSensor` in its `__init__` 

when adding entities, update is not called by default to what I understand : https://github.com/home-assistant/core/blob/87a1b8e83cb1f664863efca5e830a794864ba9c9/homeassistant/helpers/entity_platform.py#L79

but no problem to refactor as you suggest

Reply:
Done